newsfragments/3873.incompat

```diff
@@ -0,0 +1 @@
+Dropped support for Python 2.
```

Comment:
Imagining this in the NEWS file, I expect "Python 3.7 or newer is now required." probably reads better and gives a bit more information.